sapmon/payload/provider/sqlserver.py

```diff
@@ -185,25 +189,84 @@ def generateJsonString(self) -> str:
+   def _prepareSql(self,
+                   sql: str,
+                   isTimeSeries: bool,
+                   initialTimespanSecs: int) -> str:
+      self.tracer.info("[%s] preparing SQL statement" % self.fullName)
+
+      # Insert logic to get server UTC time (_SERVER_UTC)
+      sqlTimestamp = ", GETDATE() AS %s, GETUTCDATE() AS %s, GETUTCDATE() AS %s FROM" % (COL_LOCAL_UTC,COL_SERVER_UTC,COL_TIMESERIES_UTC)
+      self.tracer.debug("[%s] sqlTimestamp=%s" % (self.fullName,
+                                                  sqlTimestamp))
+      preparedSql = sql.replace(" FROM", sqlTimestamp)
+      
+      # If time series, insert time condition
+      if isTimeSeries:
+         lastRunServer = self.state.get("lastRunServer", None)
+
+         # TODO(tniek) - make WHERE conditions for time series queries more flexible
+         if not lastRunServer:
+            self.tracer.info("[%s] time series query has never been run, applying initalTimespanSecs=%d" % (self.fullName, initialTimespanSecs))
+            lastRunServerUtc = "DATEADD(s,-%d, GETUTCDATE())" % initialTimespanSecs
+         else:
+            if not isinstance(lastRunServer, datetime):
+               self.tracer.error("[%s] lastRunServer=%s could not been de-serialized into datetime object" % (self.fullName,
+                                                                                                              str(lastRunServer)))
+               return None
+            try:
+               lastRunServerUtc = "'%s'" % lastRunServer.strftime(TIME_FORMAT_SQL)
+            except Exception as e:
+               self.tracer.error("[%s] could not format lastRunServer=%s into SQL format (%s)" % (self.fullName,
+                                                                                                  str(lastRunServer),
+                                                                                                   e))
+               return None
+            self.tracer.info("[%s] time series query has been run at %s, filter out only new records since then" % \
+               (self.fullName, lastRunServerUtc))
+         self.tracer.debug("[%s] lastRunServerUtc=%s" % (self.fullName,
+                                                         lastRunServerUtc))
+         preparedSql = sql.replace("{lastRunServerUtc}", lastRunServerUtc, 1)
+         self.tracer.debug("[%s] preparedSql=%s" % (self.fullName,
+                                                    preparedSql))
+
+      # Return the finished SQL statement
+      return preparedSql
+
    # Connect to sql and run the check-specific SQL statement
-   def _actionExecuteSql(self, sql: str) -> None:
+   def _actionExecuteSql(self,
+                sql: str,
+                isTimeSeries: bool = False,
+                initialTimespanSecs: int = 60) -> None:
+
       def handle_sql_variant_as_string(value):
          return value.decode('utf-16le')
+
```

Comment:
```suggestion
      # Marking which column will be used for TimeGenerated
      self.colTimeGenerated = COL_TIMESERIES_UTC if isTimeSeries else COL_SERVER_UTC
```
We need to set `colTimeGenerated` here depending on the type of query